### PR TITLE
[feature] Support for gzip compressed files in pcap source

### LINKED ISSUE #30: Support for gzip-compressed files in pcap source
It seems to be quite common to have gzip-compressed pcap files and tools like wireshark support opening them out of the box. Similarly it would be helpful to provide the same feature for the pcap source of slimcap, as it not only extends functionality but also allows for providing compressed test files more easily (e.g. for mock tests in https://github.com/els0r/goProbe/issues/88).

### PR DESCRIPTION
**Pull request by fako1024**
Closes #30

### CODE REVIEW

File: capture/pcap/pcap_test.go [hunk 1]
```diff
@@ -248,13 +262,13 @@ func testCaptureMethods(t *testing.T, fn func(t *testing.T, src *Source)) {
 	}
 }
 
+//go:embed testdata/*
+var pcaps embed.FS
```
Comment (fako1024) [inline capture/pcap/pcap_test.go:266]: @els0r Don't know if you ever used this, but I stumbled across it while doing mock test extension on goProbe. It's a really nice and easy way to embed test files from the repository (e.g. pcap files) inside the test binary (which makes it portable) _without_ having to rely on crude mechanisms like base-64 embedding in `.go` files: https://pkg.go.dev/embed

File: capture/pcap/pcap.go [hunk 2]
```diff
@@ -35,19 +38,25 @@ func NewSource(iface string, r io.Reader) (*Source, error) {
...
+
 	// Parse the main header
 	if err := obj.read(obj.buf); err != nil {
 		return nil, err
 	}
 
 	// If required, swap endianess as defined here:
 	// https://wiki.wireshark.org/Development/LibpcapFileFormat
-	obj.header = *(*Header)(unsafe.Pointer(&obj.buf[0]))
+	obj.header = *(*Header)(unsafe.Pointer(&obj.buf[0])) // #nosec G103
```
Comment (fako1024) [inline capture/pcap/pcap.go:59]: Unrelated, but I started to audit the `unsafe` use as suggested by the Linter (basically telling it "I know what I'm doing here"), will do the same everywhere else soon.

File: capture/pcap/pcap.go [hunk 3]
```diff
@@ -35,19 +38,25 @@ func NewSource(iface string, r io.Reader) (*Source, error) {
 	if r == nil {
 		return nil, errors.New("nil io.Reader provided")
 	}
+
 	obj := Source{
-		Reader: r,
+		reader: bufio.NewReader(r),
```
Comment (fako1024) [inline capture/pcap/pcap.go:43]: Wrapped the "normal" `io.Reader` in a buffered one in order to support peeking ahead (probably also increasing performance for large files).

**Review by els0r — APPROVED** [review]
🚀

**Pull request merged**